lib/tpp/tpp_token.rb

```diff
@@ -0,0 +1,430 @@
+      end
+    end
+
+    policy = response['Policy']
+    s = policy['Subject']
+    if policy['WhitelistedDomains'].empty?
+      subjectCNRegex = [ALL_ALLOWED_REGEX]
+    else
+      if policy['WildcardsAllowed']
+        subjectCNRegex = policy['WhitelistedDomains'].map { |d| addStartEnd('[\w\-*]+' + Regexp.escape('.' + d)) }
+      else
+        subjectCNRegex = policy['WhitelistedDomains'].map { |d| addStartEnd('[\w\-]+' + Regexp.escape('.' + d)) }
+      end
+    end
+    if s['OrganizationalUnit']['Locked']
+      subjectOURegexes = escape(s['OrganizationalUnit']['Values'])
+    else
+      subjectOURegexes = [ALL_ALLOWED_REGEX]
+    end
+    if s['Organization']['Locked']
+      subjectORegexes = [escape(s['Organization']['Value'])]
+    else
+      subjectORegexes = [ALL_ALLOWED_REGEX]
+    end
+    if s['City']['Locked']
+      subjectLRegexes = [escape(s['City']['Value'])]
+    else
+      subjectLRegexes = [ALL_ALLOWED_REGEX]
+    end
+    if s['State']['Locked']
+      subjectSTRegexes = [escape(s['State']['Value'])]
+    else
+      subjectSTRegexes = [ALL_ALLOWED_REGEX]
+    end
+    if s['Country']['Locked']
+      subjectCRegexes = [escape(s['Country']['Value'])]
+    else
+      subjectCRegexes = [ALL_ALLOWED_REGEX]
+    end
+    if policy['SubjAltNameDnsAllowed']
+      if policy['WhitelistedDomains'].length == 0
+        dnsSanRegExs = [ALL_ALLOWED_REGEX]
+      else
+        dnsSanRegExs = policy['WhitelistedDomains'].map { |d| addStartEnd('[\w-]+' + Regexp.escape('.' + d)) }
+      end
+    else
+      dnsSanRegExs = []
+    end
+    if policy['SubjAltNameIpAllowed']
+      ipSanRegExs = [ALL_ALLOWED_REGEX] # todo: support
+    else
+      ipSanRegExs = []
+    end
+    if policy['SubjAltNameEmailAllowed']
+      emailSanRegExs = [ALL_ALLOWED_REGEX] # todo: support
+    else
+      emailSanRegExs = []
+    end
+    if policy['SubjAltNameUriAllowed']
+      uriSanRegExs = [ALL_ALLOWED_REGEX] # todo: support
+    else
+      uriSanRegExs = []
+    end
+
+    if policy['SubjAltNameUpnAllowed']
+      upnSanRegExs = [ALL_ALLOWED_REGEX] # todo: support
+    else
+      upnSanRegExs = []
+    end
+    unless policy['KeyPair']['KeyAlgorithm']['Locked']
+      key_types = [1024, 2048, 4096, 8192].map { |s| Vcert::KeyType.new('rsa', s) } + Vcert::SUPPORTED_CURVES.map { |c| Vcert::KeyType.new('ecdsa', c) }
+    else
+      if policy['KeyPair']['KeyAlgorithm']['Value'] == 'RSA'
+        if policy['KeyPair']['KeySize']['Locked']
+          key_types = [Vcert::KeyType.new('rsa', policy['KeyPair']['KeySize']['Value'])]
+        else
+          key_types = [1024, 2048, 4096, 8192].map { |s| Vcert::KeyType.new('rsa', s) }
+        end
+      elsif policy['KeyPair']['KeyAlgorithm']['Value'] == 'EC'
+        if policy['KeyPair']['EllipticCurve']['Locked']
+          curve = { 'p224' => 'secp224r1', 'p256' => 'prime256v1', 'p521' => 'secp521r1' }[policy['KeyPair']['EllipticCurve']['Value'].downcase]
```

Comment:
Does this break if the policy is locked to the TPP supported value of 'p384'?

Reply:
Looks like p384 is not supported right now. Should we open an issue for this?

Reply:
Yes, I will confirm what happens with it in the current release and then open an issue.

Reply:
I've opened https://github.com/Venafi/vcert-ruby/issues/17 so this issue should be addressed by a separate PR.